x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2388,241 +2386,298 @@ export class RulesClient {
-            if (apiKeyAttributes.apiKey) {
-              apiKeysMap.set(rule.id, {
-                ...apiKeysMap.get(rule.id),
-                newApiKey: apiKeyAttributes.apiKey,
-              });
-            }
+        const { apiKeyAttributes } = await this.createAndCollectApiKeys(
+          rule,
+          ruleType,
+          apiKeysMap,
+          attributes,
+          hasUpdateApiKeyOperation,
+          username
+        );
 
-            // get notifyWhen
-            const notifyWhen = getRuleNotifyWhenType(
-              attributes.notifyWhen ?? null,
-              attributes.throttle ?? null
-            );
+        const { updatedAttributes } = this.updateAttributesAndMapParams(
+          attributes,
+          apiKeyAttributes,
+          updatedParams,
+          rawAlertActions,
+          username
+        );
 
-            const updatedAttributes = this.updateMeta({
-              ...attributes,
-              ...apiKeyAttributes,
-              params: updatedParams as RawRule['params'],
-              actions: rawAlertActions,
-              notifyWhen,
-              updatedBy: username,
-              updatedAt: new Date().toISOString(),
-            });
+        rules.push({
+          ...rule,
+          references,
+          attributes: updatedAttributes,
+        });
+      } catch (error) {
+        errors.push({
+          message: error.message,
+          rule: {
+            id: rule.id,
+            name: rule.attributes?.name,
+          },
+        });
+        this.auditLogger?.log(
+          ruleAuditEvent({
+            action: RuleAuditAction.BULK_EDIT,
+            error,
+          })
+        );
+      }
+    };
+  }
+
+  private async ensureAuthorizationForBulkUpdate(
+    operations: BulkEditOperation[],
+    rule: SavedObjectsFindResult<RawRule>
+  ) {
+    for (const operation of operations) {
+      const { field } = operation;
+      if (field === 'snoozeSchedule' || field === 'apiKey') {
+        if (rule.attributes.actions.length) {
+          try {
+            await this.actionsAuthorization.ensureAuthorized('execute');
+          } catch (error) {
+            throw Error(`Rule not authorized for bulk ${field} update - ${error.message}`);
+          }
+        }
+      }
+    }
+  }
 
-            // add mapped_params
-            const mappedParams = getMappedParams(updatedParams);
+  private async getUpdatedAttributesFromOperations(
+    operations: BulkEditOperation[],
+    rule: SavedObjectsFindResult<RawRule>,
+    ruleType: RuleType
+  ) {
+    let attributes = cloneDeep(rule.attributes);
+    let ruleActions = {
+      actions: this.injectReferencesIntoActions(
+        rule.id,
+        rule.attributes.actions,
+        rule.references || []
+      ),
+    };
 
-            if (Object.keys(mappedParams).length) {
-              updatedAttributes.mapped_params = mappedParams;
+    let hasUpdateApiKeyOperation = false;
+    let isAttributesUpdateSkipped = true;
```

Comment:
Since the API request can contain multiple actions updating multiple attributes and params, `isAttributesUpdateSkipped` in plural is actually correct. The flag is set to true as long **all** of the attributes updates are skipped.